flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServer.java

```diff
@@ -396,7 +399,9 @@ public void close() throws IOException {
             if (LOG.isInfoEnabled()) {
                 LOG.info(
                         "Stopped BLOB server at {}:{}",
-                        serverSocket.getInetAddress().getHostAddress(),
+                        serverSocket == null
+                                ? null
+                                : serverSocket.getInetAddress().getHostAddress(),
                         getPort());
```

Comment:
```java
                        "Stopped BLOB server{}.",
                        serverSocket == null
                                ? ""
                                : String.format(
                                        " at %s:%d",
                                        serverSocket.getInetAddress().getHostAddress(), getPort()));
```

Reply:
I prefer to use one string interpolation method, but I noticed that `getPort` might also cause NPE,
so I added a check in in the `if` condition.

Reply:
but then you might want to add a log message in the else branch as well. Otherwise, we would miss the log message in certain scenarios.